internal/metricsharding/sharder.go

```diff
@@ -43,6 +43,30 @@ func (s *Sharder) CreateTenant(ctx context.Context, tenant TenantID, at time.Tim
 	return s.shardOpts.CreateTenant(ctx, s.yc, tenant, at)
 }
 
+// ListTenants returns list of tenants.
+func (s *Sharder) ListTenants(ctx context.Context) (result []TenantID, _ error) {
+	var (
+		lg   = zctx.From(ctx)
+		root = s.shardOpts.Root
+
+		dirs []string
+	)
+	if err := s.yc.ListNode(ctx, root, &dirs, &yt.ListNodeOptions{}); err != nil {
+		return nil, errors.Wrapf(err, "get %q dirs", root)
+	}
+
+	result = slices.Grow(result, len(dirs))
+	for _, dir := range dirs {
+		v, err := ParseTenant(dir)
```

Comment:
Probably we should just use user attributes instead of relying on naming later

Reply:
UPD: changed to tenant-1234, added user attribute with `tenant_id`: `1234` to all tables